src/common/utils/setup-helpers.js

```diff
@@ -328,18 +328,36 @@ function checkNativeTextTracksSupport(options: KPOptionsObject): void {
   }
 }
 
+/**
+ * Sets config option for Ads with MSE
+ * @private
+ * @param {KPOptionsObject} options - kaltura player options
+ * @returns {void}
+ */
+function _configureAdsWithMSE(options: KPOptionsObject): void {
+  const preferNativeHls = !!Utils.Object.getPropertyPath(options, 'playback.preferNative.hls');
+  const preferNativeDash = !!Utils.Object.getPropertyPath(options, 'playback.preferNative.dash');
+  if ((!preferNativeHls || !preferNativeDash) && options.plugins && options.plugins.ima) {
```

Comment:
1. what's about `preferNativeHls: true` but `preferNativeDash:false` (very common)
2. you can remove the native condition as you moved the attach/detach to the adapters
3. if you verify ima you have to add 1. `ima.disable:false` 2.bumper 
4. whats about dai, you can't detach/attach because the ad is part of the stream